File: frontend/src/types/category.ts
export type CategoryGroup = "plantas" | "macetas" | "quimicos";

/** Secciones del nav superior. El orden acá define el orden de los desplegables
 * y de los bloques en /categories y el footer. */
export const CATEGORY_GROUPS: { value: CategoryGroup; label: string }[] = [
  { value: "plantas", label: "Plantas" },
  { value: "macetas", label: "Macetas & Accesorios" },
  { value: "quimicos", label: "Productos Químicos" },
];

export const CATEGORY_GROUP_LABEL: Record<CategoryGroup, string> = {
  plantas: "Plantas",
  macetas: "Macetas & Accesorios",
  quimicos: "Productos Químicos",
};

export interface Category {
  category_id: string;
  name: string;
  slug: string;
  description: string | null;
  image_url: string | null;
  group: CategoryGroup;
  product_count: number;
  is_active: boolean;
  sort_order: number;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/category.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

# Sección del nav a la que pertenece la categoría. Fija en código, igual que
# los ítems del nav superior. Agregar un valor acá para sumar una sección.
CategoryGroup = Literal["plantas", "macetas", "quimicos"]

# Orden de las secciones en el nav / listados y su etiqueta visible.
CATEGORY_GROUP_ORDER: tuple[str, ...] = ("plantas", "macetas", "quimicos")
CATEGORY_GROUP_LABELS: dict[str, str] = {
    "plantas": "Plantas",
    "macetas": "Macetas & Accesorios",
    "quimicos": "Productos Químicos",
}


class CategoryOut(BaseModel):
    category_id: str
    name: str
    slug: str
    description: str | None = None
    image_url: str | None = None
    group: CategoryGroup = "plantas"
    product_count: int = 0
    is_active: bool
    sort_order: int
    created_at: datetime
    updated_at: datetime


class CategoryCreate(BaseModel):
    name: str = Field(max_length=80)
    slug: str = Field(max_length=80)
    description: str | None = Field(None, max_length=500)
    group: CategoryGroup = "plantas"
    is_active: bool = True
    sort_order: int = 0


class CategoryUpdate(BaseModel):
    name: str | None = Field(None, max_length=80)
    slug: str | None = Field(None, max_length=80)
    description: str | None = Field(None, max_length=500)
    group: CategoryGroup | None = None
    is_active: bool | None = None
    sort_order: int | None = None
